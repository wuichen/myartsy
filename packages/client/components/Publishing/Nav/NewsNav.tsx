import React from "react"
import styled from "styled-components"
import colors from "../../../Assets/Colors"
import { pMedia } from "../../Helpers"
import { NewsDateHeader, NewsText } from "../News/NewsDateHeader"

interface Props {
  date?: string
  positionTop?: number
}

export const NewsNav: React.SFC<Props> = props => {
  const { date, positionTop } = props
  return (
    <NewsNavContainer positionTop={positionTop}>
      <MaxWidthContainer>
        {date && <NewsDateHeader date={date} />}
        <Title>News</Title>
      </MaxWidthContainer>
    </NewsNavContainer>
  )
}

const Title = styled(NewsText)`
  position: absolute;
  left: 30px;
  ${pMedia.sm`
    left: 20px;
  `};
`

const MaxWidthContainer = styled.div`
  position: relative;
  max-width: 780px;
  margin: auto;
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 26px;
  ${pMedia.sm`
    min-height: 16px;
  `};
`

const NewsNavContainer = styled.div`
  position: fixed;
  top: ${(props: Props) => (props.positionTop ? props.positionTop : 0)}px;
  left: 0;
  right: 0;
  border-bottom: 1px solid ${colors.grayRegular};
  padding: 10px 0;
  background: white;
  z-index: 11;
`
