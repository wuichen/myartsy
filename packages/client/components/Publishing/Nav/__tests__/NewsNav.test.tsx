import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { NewsNav } from "../NewsNav"

describe("NewsNav", () => {
  it("renders NewsNav", () => {
    const nav = renderer.create(<NewsNav date="2017-06-29T15:00:00.000Z" />)
    expect(nav).toMatchSnapshot()
  })
})
