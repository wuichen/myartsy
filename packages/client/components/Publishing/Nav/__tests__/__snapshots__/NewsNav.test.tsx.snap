// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsNav renders NewsNav 1`] = `
.c2 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 25px;
  line-height: 1.1em;
}

.c3 {
  font-family: Unica77LLWebMedium,Arial,serif;
  -webkit-font-smoothing: antialiased;
  font-size: 25px;
  line-height: 1.1em;
  position: absolute;
  left: 30px;
}

.c1 {
  position: relative;
  max-width: 780px;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 26px;
}

.c0 {
  position: fixed;
  top: 0px;
  left: 0;
  right: 0;
  border-bottom: 1px solid #e5e5e5;
  padding: 10px 0;
  background: white;
  z-index: 11;
}

@media (max-width:720px) {
  .c2 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

@media (max-width:720px) {
  .c3 {
    font-family: Unica77LLWebMedium,Arial,serif;
    -webkit-font-smoothing: antialiased;
    font-size: 16px;
    line-height: 1.1em;
  }
}

@media (max-width:720px) {
  .c3 {
    left: 20px;
  }
}

@media (max-width:720px) {
  .c1 {
    min-height: 16px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      Jun 29, 2017
    </div>
    <div
      className="c3"
    >
      News
    </div>
  </div>
</div>
`;
